src/org/zaproxy/zap/extension/ascanrulesAlpha/TestUserAgent.java

```diff
@@ -109,14 +116,21 @@ public void scan() {
             return;
         }
 
-        for(String userAgent: USER_AGENTS){
+        for(CustomPayloadModel userAgentPayload : getUserAgentPayloads()){
             if (isStop()) {
                 return;
             }
-            attack(userAgent);
+            attack(userAgentPayload.getPayload());
         }
     }
 
+    private List<CustomPayloadModel> getUserAgentPayloads() {
+        ExtensionCustomPayloads extension = Control.getSingleton()
+                .getExtensionLoader()
+                .getExtension(ExtensionCustomPayloads.class);
```

Comment:
Should there be a null check on `extension` before trying to return? What if the extension isn't available for some reason?